### ISSUE TITLE
Migrating criteo_tft to latest version: 

### ISSUE THREAD

**Issue by picarus** [GoogleCloudPlatform/cloudml-samples#325]
**Is your feature request related to a problem? Please describe.**
I am having issues when running a piece of code using the Criteo data generated from the Criteo tft preprocess step into a piece of code I have written and that uses a newest version of tf and tft.

**Describe the solution you'd like**
I would appreciate someone can give some help on the process.

This is my input_fn function:

```
def gzip_reader_fn(filenames):
  return tf.data.TFRecordDataset(filenames, compression_type="GZIP")

def input_fn(tf_transform_output, transformed_examples, batch_size):
  """Creates an input function reading from transformed data.
  Args:
    tf_transform_output: Wrapper around output of tf.Transform.
    transformed_examples: Base filename of examples.
    batch_size: Batch size.
  Returns:
    The input function for training or eval.
  """
  def _input_fn():
    """Input function for training and eval."""
    dataset = tf.contrib.data.make_batched_features_dataset(
        file_pattern=transformed_examples,
        batch_size=batch_size,
        features=tf_transform_output.transformed_feature_spec(),
        reader=gzip_reader_fn,
        shuffle=True)
    
    #dataset = dataset.batch(batch_size) # seems to be included on the previous one
    transformed_features = dataset.make_one_shot_iterator().get_next()

    # Extract features and label from the transformed tensors.
    transformed_labels = transformed_features.pop(LABEL_KEY)

    return transformed_features, transformed_labels

  return _input_fn
```
and this is the adaptation of the [adanet example](https://github.com/tensorflow/adanet/blob/master/adanet/examples/tutorials/adanet_objective.ipynb) I am trying to get to run:

```
  def build_subnetwork(self,
                       features,
                       logits_dimension,
                       training,
                       iteration_step,
                       summary,
                       previous_ensemble=None):
    """See `adanet.subnetwork.Builder`."""
    
    if KEY_FEATURE_COLUMN in features:
        features.pop(KEY_FEATURE_COLUMN)
 
    input_layer = tf.to_float(features.values())
    #print(input_layer.shape)
    input_layer = tf.transpose(input_layer,[1,0])  # this I don't understand why I need it, but without it the code does not work.
    #print(input_layer.shape)
    kernel_initializer = tf.glorot_uniform_initializer(seed=self._seed)
    last_layer = input_layer
    for _ in range(self._num_layers):
```

Any hints on the reason "transpose" is needed? Without it I get an error saying:
logits and labels must have the same shape ((39, 1) vs (512, 1))

Where 39 is the number of features and 512 my batch size.

The data was generated with the original preprocess.py in criteo_tft.
 
I will contribute the code if I get to work, maybe you can add another example with Adanet.
I attach the code in case it helps to diagnose the issue.

[criteo_tft-adanet.zip](https://github.com/GoogleCloudPlatform/cloudml-samples/files/2632460/criteo_tft-adanet.zip)

**Comment by happyhuman** [comment]
Hi. Thanks for opening the issue. Based on your description, it seems your question is mostly a tensorflow related issue, and not a Google Cloud related issue. Is this the case?

**Comment by picarus** [comment]
I think is an integration problem or a backward compatibility issue, between tft using in the criteo tft preprocess step to create the TFRecordDataset and tft and tf used in my code to read it.

**Comment by happyhuman** [comment]
Since our priority in this repo is to show how to use Google Cloud Platform and its products, we may not be able to address your tensorflow related question in a timely manner. May I suggest that you contact tensorflow support team for your TF related question?

**Comment by picarus** [comment]
@happyhuman, the dataflow product is made of tf, tft and apache-beam (and probably a few others) so I think the issue is relevant to this group. I am finding very complicated getting anything done within GCP because of many version incompatibilities. Only the tft repo has [info about versions](https://github.com/tensorflow/transform) but then your samples are way behind. I am trying to make adanet work within GCP and help updating versions, as your colleague @gogasca pointed out when I opened [this other issue](https://github.com/GoogleCloudPlatform/cloudml-samples/issues/314) from which he created [this feature request](https://github.com/GoogleCloudPlatform/cloudml-samples/issues/319).

So yes, I will raise my issue in tft repo, but I think from this repo you should be able to help because the GCP products you are trying to explain how they work, are based on these components we are having issues with. 
 
A review of your samples for every major release of tf I think would be the minimum to have, dropping the samples that don't work or documenting the issues.
Thanks to @gogasca for his help.

**Comment by gogasca** [comment]
We decided not to upgrade legacy samples, we are working in modularizing the samples to show functionality of GCP ML products. If you have need specific functionality please open a new issue. Thanks
